### ISSUE TITLE
Best Wishes; Use My Own Dataset ?

### ISSUE THREAD

**Issue by kpoots** [hosseinfani/ReQue#6]
Hope you are doing well. This is the first week of school I think - so best wishes and do your best !
I am sure that you will do fine.

I would like to use my own dataset, which is a TREC dataset including corpus and qrels. 
I am partway through trying to mimic the format of a dataset which is used by ReQue. 
I need to sort-out the qrels portion of the dataset. 

Is there a better means of using my own dataset ? Did I miss some notes ?
Thank you !

**Comment by hosseinfani** [comment]
Thanks for your kind message.

1) ReQue needs the corpus to be indexed for the task of retrieval (search). So you have to index your corpus by command like [this](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L10) that we used for robust04.

2) Then, given the query file e.g., q.txt and using an expander like wordnet, it creates a new file q.wordnet.txt [here](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L46).

3) Then it searches the expanded queries in q.wordnet.txt in the indexed corpus [here](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L64). As you can see, ReQue depends on anserini for retrieval. If anserini does not support that, a new retrieval should be developed. Finally, it stores the result in a new file q.wordnet.bm25.txt if bm25 is used for retrieval. This file has the docid of the relevant docs in your corpus.

4) Finally, ReQue evaluates the results in q.wordnet.bm25.txt using your qrels [here](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L86). ReQue uses trec_eval so you can use any metrics supported therein. If you use MAP, it stores the result in a new file q.wordnet.bm25.map.txt for each query in the q.wordnet.txt.

A sample flow for robust04 is from [here ](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L175)to [here](https://github.com/hosseinfani/ReQue/blob/df3bcbdc3189a936f39ade0743450b7871e35517/qe/main.py#L177).
In short, so far ReQue does not support a corpus that cannot be indexed or searched using anserini.
Hope it answers your question.

**Comment by Narabzad** [comment]
@kpoots This might be useful as well in case you want to index your own dataset 
 https://github.com/castorini/pyserini#how-do-i-search-my-own-documents

**Comment by kpoots** [comment]
Thank you both !  Please take care.
